### PR TITLE
Add a `testcontainers_wiremock#RunContainerAndStopOnCleanup()` method to simplify automatic termination

### PR DESCRIPTION
**Pull request by Vayras**
As per the issue I have containerize the code the update code is in a new branch with same name as the issue
## References

- TODO

## Submitter checklist

- [x] The PR request is well described and justified, including the body and the references
- [x] The PR title represents the desired changelog entry
- [ ] The repository's code style is followed (see the contributing guide)
- [ ] Test coverage that demonstrates that the change works as expected
- [x] For new features, there's necessary documentation in this pull request or in a subsequent PR to [wiremock.org](https://github.com/wiremock/wiremock.org)

Screenshot 2023-08-09 172259

### CODE REVIEW

**Review by oleg-nenashev — COMMENTED** [review]
Thanks for the pull request. I would actually expect not refactoring of a test case, but adding a new method in https://github.com/wiremock/wiremock-testcontainers-go/blob/main/tc-wiremock.go that makes  the test more simple 

It could be `RunContainerAndStopOnCleanup()` or so that takes the testing context as an extra argument

**Comment by Vayras** [comment]
@oleg-nenashev thanks for your feedback , correct me if I'm wrong , the method needs to written in the _tc-wiremock.go_ and used in the _quickstart_test.go_?

**Comment by oleg-nenashev** [comment]
@Vayras yup, this is correct. See https://github.com/wiremock/wiremock-testcontainers-go/pull/21 for inspiration

**Comment by Vayras** [comment]
@oleg-nenashev should I implement these changes?

**Comment by oleg-nenashev** [comment]
@Vayras this is an expectation in #27

**Comment by Vayras** [comment]
@oleg-nenashev I have commited and pushed some changes using terminal can you please let me know if they reflect and are updated?

File: tc-wiremock.go [hunk 1]
```diff
@@ -180,3 +181,21 @@ func addQueryParamsToURL(endpoint string, queryParams map[string]string) (string
 
 	return parsedURL.String(), nil
 }
+
+func RunContainerAndStopOnCleanup(ctx context.Context, t *testing.T, mappingFileName string) (testcontainers.Container, error) {
+	container, err := RunContainer(ctx,
+		WithMappingFile("hello", mappingFileName),
```
Comment (oleg-nenashev) [inline tc-wiremock.go:187]: That part needs to be parameterizable and, preferably, taking the same array as a `RunContainer` method

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Thank you! It gets closer but we need to replace the `/hello` hardcoding by a proper mapping of parameters

File: tc-wiremock.go (hunk 1 above)
Comment (Vayras) [reply tc-wiremock.go:187]: @oleg-nenashev Thank you for your feedback , will make these changes

Comment (Vayras) [reply tc-wiremock.go]: Screenshot 2023-08-15 224903
I think by adding an Id parameter the issue will be fixed

**Review by oleg-nenashev — COMMENTED** [review]
@Vayras I suggest keeping the original `RunContainerSomething(ctx context.Context, opts ...testcontainers.ContainerCustomizer)` interface so that the code is more flexible. You would need to just take this array as an argument and pass to the nested method

**Comment by oleg-nenashev** [comment]
Thanks! Let's see how the CI runs, and happy to integrate it afterwards

**Review by oleg-nenashev — APPROVED** [review]
Thanks! I will update the README in the downstream PR, because there are a few other changes I wanted to land anyway

**Comment by Vayras** [comment]
@oleg-nenashev It's truly been a great experience collaborating with you on this project. If the thought crosses your mind, I'd be honored to discuss the possibility of sponsorship for my work or even the potential of becoming a part of this organization. Please feel free to reach out anytime. I'm always eager to contribute my skills and expertise to address challenges as effectively as I can.

**Comment by oleg-nenashev** [comment]
@Vayras from what I understand, you're just starting with Golang. I might be open to discuss various open source internship options in the future, but at the same time I want to highlight that asking for being paid for your work in the future after doing a single newcomer friendly contribution is unlikely to be perceived well by maintenance. So I recommend to revise the message above if you send it to other projects

**Comment by Vayras** [comment]
@oleg-nenashev Thank you for addressing this  , finding an open source internship related to Golang would be very beneficial for me, and I think open source contribution , even with whatever little I do , keeps me motivated ,I'm fairly new to Golang , I have recently picked up learning this language any opportunity to get better at it is deeply appreciated.

**Pull request merged**